### PR TITLE
Prevent writes to free'd JFR buffer

### PR DESCRIPTION
**Pull request by tajila**
Also, check that threadObject is not NULL before dereference.

### CODE REVIEW

**Comment by tajila** [comment]
jenkins test sanity.functional xlinux jdk11

**Comment by tajila** [comment]
@gacholio Please review these changes

**Review by gacholio — APPROVED** [review]

**Comment by gacholio** [comment]
```
11:10:48  /usr/local/gcc11/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: ../lib/libomrutil.a(thrname_core.c.o): in function `getOMRVMThreadNameNoLock':
11:10:48  /home/jenkins/workspace/Build_JDK21_x86-64_linux_Personal/omr/util/omrutil/thrname_core.c:36: undefined reference to `OMR_Glue_GetThreadNameForUnamedThread'
11:10:48  /usr/local/gcc11/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /home/jenkins/workspace/Build_JDK21_x86-64_linux_Personal/omr/util/omrutil/thrname_core.c:36: undefined reference to `OMR_Glue_GetThreadNameForUnamedThread'
11:10:48  [ 28%] Building CXX object runtime/gc_vlhgc/CMakeFiles/j9gcvlhgc.dir/CompactGroupPersistentStats.cpp.o
11:10:48  collect2: error: ld returned 1 exit status
11:10:48  gmake[6]: *** [runtime/libj9trc29.so] Error 1
```

**Comment by tajila** [comment]
I may need to rebase my changes

**Comment by tajila** [comment]
jenkins compile xlinux jdk21

File: runtime/vm/jfr.cpp [hunk 1]
```diff
@@ -180,7 +198,7 @@ flushBufferToGlobal(J9VMThread *currentThread, J9VMThread *flushThread)
 	UDATA bufferSize = flushThread->jfrBuffer.bufferCurrent - flushThread->jfrBuffer.bufferStart;
 	bool success = true;
 
-	if (NULL == flushThread->jfrBuffer.bufferStart) {
+	if (areJFRBuffersReadyForWrite(currentThread)) {
```
Comment (keithc-ca) [inline runtime/vm/jfr.cpp:201]: Is this missing `!`? Shouldn't it be
```c
	if (!areJFRBuffersReadyForWrite(currentThread)) {
```

File: runtime/vm/jfr.cpp [hunk 2]
```diff
@@ -284,7 +302,9 @@ reserveBuffer(J9VMThread *currentThread, UDATA size)
 	Assert_VM_true(((currentThread)->publicFlags & J9_PUBLIC_FLAGS_VM_ACCESS)
 	|| ((J9_XACCESS_EXCLUSIVE == vm->exclusiveAccessState) || (J9_XACCESS_EXCLUSIVE == vm->safePointState)));
 
-
+	if (areJFRBuffersReadyForWrite(currentThread)) {
```
Comment (keithc-ca) [inline runtime/vm/jfr.cpp:305]: Missing `!`?

Comment (tajila) [reply runtime/vm/jfr.cpp:305]: yes, https://github.com/eclipse-openj9/openj9/pull/20621

**Pull request merged**